tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/QuantRule.scala

```diff
@@ -73,10 +75,18 @@ class QuantRule(rewriter: SymbStateRewriter) extends RewritingRule with LazyLogg
     }
   }
 
-  private def isAssignmentInside(ex: TlaEx): Boolean = ex match {
-    case OperEx(ApalacheOper.assign, OperEx(TlaActionOper.prime, NameEx(_)), _) => true
-    case OperEx(_, args @ _*)                                                   => args.exists(isAssignmentInside)
-    case _                                                                      => false
+  private def findAssignedNames(ex: TlaEx): Map[String, TlaType1] = ex match {
+    case OperEx(ApalacheOper.assign, OperEx(TlaActionOper.prime, nex @ NameEx(name)), _) =>
+      Map(name -> nex.typeTag.asTlaType1)
+
+    case OperEx(_, args @ _*) =>
+      args.flatMap(findAssignedNames).toMap
+
+    case LetInEx(body, decls @ _*) =>
+      decls.flatMap(d => findAssignedNames(d.body)).toMap ++ findAssignedNames(body)
```

Comment:
```suggestion
      decls.foldLeft(findAssignedNames(body)) { case (map, d) => map ++ findAssignedNames(d.body)  }
```

Reply:
This is not a correctness bug btw, more of a style preference

Reply:
I think Intellij suggested to replace your variant :)